Bad error message when `pnpm licenses list` fails

When not running `pnpm install` beforehand, we can get the following error

```
+ pnpm-licenses generate-disclaimer --prod --output-file=../THIRDPARTY-frontend.yml
file://$BUILD_PREFIX/lib/node_modules/@quantco/pnpm-licenses/dist/chunk-S3XV4FTF.mjs:83
          return reject(new Error(stderr));
                       ^
Error
    at file://$BUILD_PREFIX/lib/node_modules/@quantco/pnpm-licenses/dist/chunk-S3XV4FTF.mjs:83:25
    at ChildProcess.exithandler (node:child_process:427:5)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1104:16)
    at ChildProcess._handle.onexit (node:internal/child_process:304:5)
Node.js v22.11.0
```

which refers to the following code part in pnpm-licenses

```
      exec(`pnpm licenses list ${options.prod ? "--prod" : ""} --json`, (error, stdout, stderr) => {
        if (error)
          return reject(new Error(stderr));
        resolve(stdout);
      });
```

running `pnpm licenses list` returns the following error

```
+ pnpm licenses list --prod --json
file://$BUILD_PREFIX/lib/node_modules/@quantco/pnpm-licenses/dist/chunk-S3XV4FTF.mjs:71
  throw new Error(`${code} - ${message}${stringifiedRest}`);
        ^
Error: ERR_PNPM_MISSING_PACKAGE_INDEX_FILE - Failed to find package index file for uplot@1.6.31, please consider running 'pnpm install'
    at checkForPnpmError (file://$BUILD_PREFIX/lib/node_modules/@quantco/pnpm-licenses/dist/chunk-S3XV4FTF.mjs:71:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async generateDisclaimerCommand (file://$BUILD_PREFIX/lib/node_modules/@quantco/pnpm-licenses/dist/index.mjs:38:66)
Node.js v22.11.0
```

It would be nice if pnpm could print out stdout/stderr of the failing subprocess

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -6,12 +6,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: pnpm/action-setup@v2.2.4
+      - uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2
         with:
-          version: 6.32.2
+          version: 9.14.4
       - uses: actions/setup-node@v3
         with:
-          node-version: "18"
+          node-version: "22"
           cache: "pnpm"
       - run: pnpm install
 

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -20,13 +20,13 @@ jobs:
       packages: write
     steps:
       - uses: actions/checkout@v3
-      - uses: pnpm/action-setup@v2.2.4
+      - uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2
         with:
-          version: 6.32.2
+          version: 9.14.4
       # Setup .npmrc file to publish to GitHub Packages
       - uses: actions/setup-node@v3
         with:
-          node-version: "18"
+          node-version: "22"
           cache: "pnpm"
           registry-url: ${{ matrix.packages.registry }}
           scope: ${{ matrix.packages.scope }}

--- a/LICENSE
+++ b/LICENSE
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2023 Jannik Wibker
+Copyright (c) 2023-2024 Janina Wibker
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@quantco/pnpm-licenses",
-  "version": "2.1.0",
+  "version": "2.2.0",
   "description": "Generate third party license disclaimers in pnpm-based projects",
   "homepage": "https://github.com/Quantco/pnpm-licenses",
   "repository": {
@@ -24,7 +24,7 @@
     "pnpm",
     "cli"
   ],
-  "author": "Jannik Wibker",
+  "author": "Janina Wibker",
   "license": "MIT",
   "scripts": {
     "build": "tsup",

--- a/src/cli.ts
+++ b/src/cli.ts
@@ -204,7 +204,10 @@ if (argv._.length === 1 && argv._[0] === 'list') {
     process.exit(1)
   }
 
-  listCommand({ prod: argv.prod, filters }, ioOptions)
+  listCommand({ prod: argv.prod, filters }, ioOptions).catch((error) => {
+    console.log(error)
+    process.exit(1)
+  })
 }
 
 if (argv._.length === 1 && argv._[0] === 'generate-disclaimer') {
@@ -253,5 +256,8 @@ if (argv._.length === 1 && argv._[0] === 'generate-disclaimer') {
     process.exit(1)
   }
 
-  generateDisclaimerCommand({ prod: argv.prod, filters }, ioOptions)
+  generateDisclaimerCommand({ prod: argv.prod, filters }, ioOptions).catch((error) => {
+    console.log(error)
+    process.exit(1)
+  })
 }

--- a/src/get-dependencies.ts
+++ b/src/get-dependencies.ts
@@ -112,7 +112,7 @@ export const getDependencies = (
   } else {
     inputPromise = new Promise((resolve, reject) => {
       exec(`pnpm licenses list ${options.prod ? '--prod' : ''} --json`, (error, stdout, stderr) => {
-        if (error) return reject(new Error(stderr))
+        if (error) return reject(new Error(`${error.message}\nstdout: ${stdout}\nstderr: ${stderr}`))
         resolve(stdout)
       })
     })

